resources/aws_ec2.go

```diff
@@ -802,16 +808,21 @@ func (obj *AwsEc2Res) prependName() string {
 	return AwsPrefix + obj.GetName()
 }
 
-// snsServer returns an http server used to listen for sns messages.
-func (obj *AwsEc2Res) snsServer() *http.Server {
-	addr := obj.WatchListenAddr
-	// if addr is a port
+func (obj *AwsEc2Res) snsListener(addr string) (net.Listener, error) {
 	if _, err := strconv.Atoi(obj.WatchListenAddr); err == nil {
 		addr = fmt.Sprintf(":%s", obj.WatchListenAddr)
 	}
-	handler := http.HandlerFunc(obj.snsPostHandler)
+	listener, err := net.Listen("tcp", addr)
+	if err != nil {
+		return nil, err
+	}
+	return listener, nil
+}
+
+// snsServer returns an http server used to listen for sns messages.
```

Comment:
I agree with you now that this might be overkill for a phone function. You can kill it if you'd like.